core/testing/src/main/kotlin/com/google/samples/apps/nowinandroid/core/testing/repository/TestRecentSearchRepository.kt

```diff
@@ -19,14 +19,16 @@ package com.google.samples.apps.nowinandroid.core.testing.repository
 import com.google.samples.apps.nowinandroid.core.data.model.RecentSearchQuery
 import com.google.samples.apps.nowinandroid.core.data.repository.RecentSearchRepository
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.flow
 
 class TestRecentSearchRepository : RecentSearchRepository {
 
     private val cachedRecentSearches: MutableList<RecentSearchQuery> = mutableListOf()
 
     override fun getRecentSearchQueries(limit: Int): Flow<List<RecentSearchQuery>> =
-        flowOf(cachedRecentSearches.sortedByDescending { it.queriedDate }.take(limit))
```

Comment:
is this change necessary?

Reply:
yes, i think that this code need to be changed. because the flow producers such as 'flowOf' and 'flow { emit() } have different in emitting updated list. and although one depth inner code is samely emitting data by 'flow { emit() }, two depth inner code is different. because of this, i realized that the way of data emitting can not be same.

**[two depth inner code]**
![스크린샷 2024-04-04 오전 10 49 00](https://github.com/android/nowinandroid/assets/57611838/dfa44462-84e9-4a10-9c06-1c5de8afa29d)
 **left : flow { emit() } / right : flowOf()**
[additional reference, my blog column(lang korean) : different between flowOf and flow { emit() }](https://velog.io/@squart300kg/flow%EC%99%80-flowOf-%EC%B0%A8%EC%9D%B4-%EB%AA%A8%EB%A5%B4%EB%A9%B4-%EB%B0%A4%EC%83%98-%EC%95%BC%EA%B7%BC%ED%95%A0-%EC%88%98-%EC%9E%88%EB%8B%A4)

because of this change, i had tested 'SearchViewModelTest' and passed. so, i think that this change is reasonable. how about your opinion?

Reply:
Sorry, I don't understand this response. `flowOf` is used to construct a flow from the list of search queries and emit the list immediately. I don't see any benefit to using a `flow` builder here as that usually implies more complex logic. But maybe I'm missing something?